### PR TITLE
Add Configuration Option and Filter to mute Elasticsearch Deprecation Warnings

### PR DESCRIPTION
**Pull request by janheise**
## Description
This change adds an interceptor to HttpClient that filters deprecation warnings from the Http-response headers if the configuration option is set.

Fixes for #10163 and #9822

## Motivation and Context
The Elasticsearch deprecation warnings might generate bloat in the logs that are not helpful.

## How Has This Been Tested?
Manual tests and an automatic test class for a working case and for corner cases

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-elasticsearch7/src/test/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptorTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+public class ElasticsearchFilterDeprecationWarningsInterceptorTest {
+    @Test
+    public void testInterceptorNoHeader() throws IOException, HttpException {
+        ElasticsearchFilterDeprecationWarningsInterceptor interceptor = new ElasticsearchFilterDeprecationWarningsInterceptor();
+
+        HttpResponse response = new BasicHttpResponse(new BasicStatusLine(new ProtocolVersion("HTTP", 0, 0), 0, null));
+        interceptor.process(response, null);
+
+        Assert.assertEquals( "Number of Headers should be 0 and the interceptor should not fail in itself.", 0, response.getAllHeaders().length);
```
Comment (dennisoelkers) [inline graylog-storage-elasticsearch7/src/test/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptorTest.java:37]: We prefer to use [AssertJ](https://assertj.github.io/doc/) for assertions, so this would look like this:
```suggestion
        assertThat(response.getAllHeaders())
                .as("Number of Headers should be 0 and the interceptor should not fail in itself.")
                .hasSize(0);
```

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ElasticsearchFilterDeprecationWarningsInterceptor implements HttpResponseInterceptor {
+    @Override
+    public void process(HttpResponse response, HttpContext context) throws
+            HttpException, IOException {
+        List<Header> warnings = Arrays.stream(response.getHeaders("Warning")).filter(header -> !header.getValue().contains("setting was deprecated in Elasticsearch")).collect(Collectors.toList());
```
Comment (dennisoelkers) [inline graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptor.java:34]: There are some more deprecation warnings that are probably not catched by this, that we may or may not want to mute. One example is this: https://github.com/Graylog2/graylog2-server/issues/9822 (with a helpful link to the [source]( https://github.com/elastic/elasticsearch/blob/1a05a5ac24222d3e00a7f1679b12e700004af534/server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/TransportGetAliasesAction.java#L129))

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good, added some suggestions inline!

File: graylog-storage-elasticsearch7/src/test/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptorTest.java (hunk 1 above)
Comment (janheise) [reply graylog-storage-elasticsearch7/src/test/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptorTest.java]: fixed it like suggested

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptor.java (hunk 2 above)
Comment (janheise) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptor.java]: More warnings have been added to the filter

**Review by dennisoelkers — APPROVED** [review]
Good work! :tada:

**Comment by liujunit** [comment]
When elasticsearch sets x-pack security authentication, you need to log in with an account and password,
When elasticsearch_mute_deprecation_warnings = true
```
if(muteElasticsearchDeprecationWarnings) {
    restClientBuilder.setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.addInterceptorFirst(new ElasticsearchFilterDeprecationWarningsInterceptor()));
}
```
It rewrites the previous settings，but it does not set the DefaultCredentialsProvider
**Causing an error: missing authentication credentials for rest request**，
Should it be set like this ?：
```
if(muteElasticsearchDeprecationWarnings) {
            restClientBuilder.setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.addInterceptorFirst(new ElasticsearchFilterDeprecationWarningsInterceptor())
                .setMaxConnTotal(maxTotalConnections)
                .setMaxConnPerRoute(maxTotalConnectionsPerRoute)
                .setDefaultCredentialsProvider(credentialsProvider)
            );
 }
```
My English is poor~~~QAQ
